src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/ClassReader.java

```diff
@@ -3345,7 +3345,8 @@ long adjustClassFlags(ClassSymbol c, long flags) {
             flags &= ~ACC_MODULE;
             flags |= MODULE;
         }
-        if (((flags & ACC_IDENTITY) != 0 && !isMigratedValueClass(flags)) || (majorVersion < V67.major && (flags & INTERFACE) == 0)) {
+        if (((flags & ACC_IDENTITY) != 0 && !isMigratedValueClass(flags))
```

Comment:
do we still need the `isMigratedValueClass` ? Aren't migrated value classes treated the same as ordinary value classes thanks to the work we did to load differerent classfiles based on version?

Reply:
I only adjusted this because this is incorrectly marking Java 25 classes as value classes in InnerClasses. This call didn't affect my patch, should I investigate it now?

Reply:
It's ok to investigate in a separate PR